File: 04. Fall 2024/CS 4348 - Operating Systems Concepts/Project 01/Logger.java
import java.io.*;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Logger {
    
    public static void main(String[] args) {
        
        if (args.length != 1) {
            System.out.println("Usage: java Logger <logFileName>");
            return;
        }

        String logFileName = args[0];
        
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(logFileName, true));
             BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
            
            String line;
            while ((line = reader.readLine()) != null) {
                
                if ("QUIT".equals(line.trim())) {
                    break;
                }
                
                String timestamp = new SimpleDateFormat("yyyy-MM-dd HH:mm").format(new Date());
                writer.write(String.format("%s [%s]\n", timestamp, line));
                writer.flush();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: 04. Fall 2024/CS 4348 - Operating Systems Concepts/Project 01/logger.py
# logger.py
import sys
from datetime import datetime

def log_message(log_file, action, message):
    with open(log_file, 'a') as f:
        timestamp = datetime.now().strftime('%Y-%m-%d %H:%M')
        f.write(f"{timestamp} [{action}] {message}\n")

def main(log_file):
    while True:
        line = sys.stdin.readline().strip()
        if line == "QUIT":
            break
        action, message = line.split(" ", 1)
        log_message(log_file, action, message)

if __name__ == "__main__":
    log_file = sys.argv[1]
    main(log_file)